packages/design-system-react-native/src/components/BadgeIcon/BadgeIcon.test.tsx

```diff
@@ -0,0 +1,203 @@
+import React from 'react';
+import { render } from '@testing-library/react-native';
+import { useTailwind } from '@metamask/design-system-twrnc-preset';
+
+import Text from '../Text';
+import { IconName } from '../Icon';
+import BadgeIcon from './BadgeIcon';
+import {
+  MAP_BADGEICON_VARIANT_ICONNAME,
+  DEFAULT_BADGEICON_PROPS,
+} from './BadgeIcon.constants';
+import { BadgeIconVariant } from './BadgeIcon.types';
+
+describe('BadgeIcon', () => {
+  it('renders with variant Custom and forwards provided iconName and iconProps', () => {
+    const TestComponent = () => {
+      const tw = useTailwind();
+      // Compute expected container style using an empty twClassName.
+      const computedExpectedContainer = tw`
+        h-[16px] w-[16px] bg-icon-default rounded-full items-center justify-center`;
+      return (
+        <>
+          <BadgeIcon
+            variant={BadgeIconVariant.Custom}
+            iconName={IconName.Add}
+            testID="badge-icon"
+          />
+          <Text testID="expectedContainer">
+            {JSON.stringify(computedExpectedContainer)}
+          </Text>
+        </>
+      );
+    };
+
+    const { getByTestId } = render(<TestComponent />);
+    const expectedContainer = JSON.parse(
+      getByTestId('expectedContainer').props.children,
+    );
+    const badgeIcon = getByTestId('badge-icon');
+    // Verify container style.
+    expect(badgeIcon.props.style[0]).toStrictEqual(expectedContainer);
+    // Verify Icon receives the provided custom iconName.
+    const icon = badgeIcon.props.children;
+    expect(icon.props.name).toStrictEqual('Add');
+    expect(icon.props.color).toStrictEqual(
+      DEFAULT_BADGEICON_PROPS.iconProps.color,
+    );
+    expect(icon.props.size).toStrictEqual(
+      DEFAULT_BADGEICON_PROPS.iconProps.size,
+    );
+  });
+
+  it('renders with variant Snaps and uses mapping for final iconName', () => {
+    const TestComponent = () => {
+      const tw = useTailwind();
+      const computedExpectedContainer = tw`
+        h-[16px] w-[16px] bg-icon-default rounded-full items-center justify-center`;
```

Comment:
Suggestion: Can we start using the [default Tailwind sizing system](https://v3.tailwindcss.com/docs/customizing-spacing#default-spacing-scale) when available for class names? I mentioned in a previous PR review that while there won’t be performance benefits for `twrnc` but we should see some for web components and it would be good to be consistent across platform and align with our [spacing tokens](https://github.com/MetaMask/metamask-design-system/issues/157)

```suggestion
        h-4 w-4 bg-icon-default rounded-full items-center justify-center`;
```